src/scenes/MyAccountRouter/MyAccountView.tsx

```diff
@@ -205,6 +208,7 @@ const MyAccountContent = ({
     setPreferredWeightSystemSelected((userPreferences?.preferredWeightSystem as string) || 'metric');
     setLocaleSelected(selectedLocale);
     setSelectedRows([]);
+    onChange('cookiesConsented', user.cookiesConsented);
```

Comment:
Shouldn't the value of this already be set? It seems like the change on 555 should be sufficient.

Reply:
This resets the form value when cancel is pressed.

Reply:
But aren't you navigating away from the form? Is the form state still there if you navigate away and come back to the form?

Reply:
I'm not sure exactly, but this change was needed otherwise the value stayed modified after pressing cancel in the form.